Tests/src/artofillusion/object/SceneCameraTest.java

```diff
@@ -0,0 +1,484 @@
+  public void createSceneCameraFromStreamBadFilter() throws IOException {
+    ByteBuffer wrap = ByteBuffer.allocate(100);
+    wrap.putShort((short)1); // Base Object3d version
+    wrap.putShort((short)3); // Scene Camera object version
+    wrap.putDouble(123); // distToPlane
+    wrap.putDouble(45); // fov
+    wrap.putDouble(100); // depthOfFieled
+    wrap.putDouble(50); // focalDistance;
+    wrap.put((byte)1);  // Perspective Boolean treats as byte 
+    wrap.putInt(1); // No camera filters
+    
+    String filterName = "UnknownFilterClass";
+    byte[] className = filterName.getBytes("UTF-8");
+    Integer  cnl = className.length;
+    wrap.putShort(cnl.shortValue());
+    wrap.put(className);
+    
+    SceneCamera camera = new SceneCamera(StreamUtil.stream(wrap), null);
+    Assert.assertEquals(45f, camera.getFieldOfView(), 0);
+    Assert.assertEquals(123, camera.getDistToPlane(), 0);
+    Assert.assertTrue(camera.isPerspective());
+    Assert.assertEquals(100f, camera.getDepthOfField(), 0);
+    Assert.assertEquals(50f, camera.getFocalDistance(), 0);
+    Assert.assertEquals(0, camera.getImageFilters().length);
+  }  
+  
+  
+  
+  @Test
+  public void сreateSceneCameraFromStreamSingleGoodFilter() throws IOException {
+    ByteBuffer wrap = ByteBuffer.allocate(200);
+    wrap.putShort((short)1); // Base Object3d version
+    wrap.putShort((short)3); // Scene Camera object version
+    wrap.putDouble(123); // distToPlane
+    wrap.putDouble(45); // fov
+    wrap.putDouble(100); // depthOfFieled
+    wrap.putDouble(50); // focalDistance;
+    wrap.put((byte)1);  // Perspective Boolean treats as byte 
+    wrap.putInt(1); //One Camera filter
+    
+    
+    String filterName = DummyImageFilter.class.getName();
+    
+    byte[] className = filterName.getBytes("UTF-8");
+    Integer  cnl = className.length;
+    wrap.putShort(cnl.shortValue());
+    wrap.put(className);
+    
+    
+    
+    SceneCamera camera = new SceneCamera(StreamUtil.stream(wrap), null);
+    Assert.assertEquals(45f, camera.getFieldOfView(), 0);
+    Assert.assertEquals(123, camera.getDistToPlane(), 0);
+    Assert.assertTrue(camera.isPerspective());
+    Assert.assertEquals(100f, camera.getDepthOfField(), 0);
+    Assert.assertEquals(50f, camera.getFocalDistance(), 0);
+    Assert.assertEquals(1, camera.getImageFilters().length);
+    Assert.assertTrue(camera.getImageFilters()[0] instanceof DummyImageFilter);
+  }  
+  
+  @Test
+  public void сreateSceneCameraFromStreamDoubleGoodFilter() throws IOException {
+    ByteBuffer wrap = ByteBuffer.allocate(200);
+    wrap.putShort((short)1); // Base Object3d version
+    wrap.putShort((short)3); // Scene Camera object version
+    wrap.putDouble(123); // distToPlane
+    wrap.putDouble(45); // fov
+    wrap.putDouble(100); // depthOfFieled
+    wrap.putDouble(50); // focalDistance;
+    wrap.put((byte)1);  // Perspective Boolean treats as byte 
+    wrap.putInt(2); //One Camera filter
```

Comment:
Do we really want to be exercising the format-version code with hand-constructed byte streams? Is there a better way to test this, or should be be bothering at all?

@peastman, I'd appreciate some input.